packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts

```diff
@@ -40,74 +40,121 @@ export interface BuilderOptions {
   /**
    * The node version to use as target for Babel
    */
-  readonly nodeVersion?: string;
+  readonly nodeVersion: string;
+
+  /**
+   * The docker tag of the node base image to use in the parcel-bundler docker image
+   *
+   * @see https://hub.docker.com/_/node/?tab=tags
+   */
+  readonly nodeDockerTag: string;
 }
 
 /**
  * Builder
  */
 export class Builder {
-  private readonly parcelBinPath: string;
+  private readonly pkgPath: string;
 
-  constructor(private readonly options: BuilderOptions) {
-    let parcelPkgPath: string;
-    try {
-      parcelPkgPath = require.resolve('parcel-bundler/package.json'); // This will throw if `parcel-bundler` cannot be found
-    } catch (err) {
-      throw new Error('It looks like parcel-bundler is not installed. Please install v1.x of parcel-bundler with yarn or npm.');
-    }
-    const parcelDir = path.dirname(parcelPkgPath);
-    const parcelPkg = JSON.parse(fs.readFileSync(parcelPkgPath, 'utf8'));
+  private readonly originalPkg: Buffer;
 
-    if (!parcelPkg.version || !/^1\./.test(parcelPkg.version)) { // Peer dependency on parcel v1.x
-      throw new Error(`This module has a peer dependency on parcel-bundler v1.x. Got v${parcelPkg.version}.`);
-    }
+  private readonly originalPkgJson: { [key: string]: any };
 
-    this.parcelBinPath = path.join(parcelDir, parcelPkg.bin.parcel);
+  constructor(private readonly options: BuilderOptions) {
+    // Original package.json
+    this.pkgPath = findPkgPath();
+    this.originalPkg = fs.readFileSync(this.pkgPath);
+    this.originalPkgJson = JSON.parse(this.originalPkg.toString());
   }
 
+  /**
+   * Build with parcel in a Docker container
+   */
   public build(): void {
-    const pkgPath = findPkgPath();
-    let originalPkg;
-
     try {
-      if (this.options.nodeVersion && pkgPath) {
-        // Update engines.node (Babel target)
-        originalPkg = updatePkg(pkgPath, {
-          engines: { node: `>= ${this.options.nodeVersion}` },
-        });
+      this.updatePkg();
+
+      const dockerBuildArgs = [
+        'build',
+        '--build-arg', `NODE_TAG=${this.options.nodeDockerTag}`,
+        '-t', 'parcel-bundler',
+        path.join(__dirname, '../parcel-bundler'),
+      ];
+
+      const build = spawnSync('docker', dockerBuildArgs);
+
+      if (build.error) {
+        throw build.error;
+      }
+
+      if (build.status !== 0) {
+        throw new Error(`[Status ${build.status}] stdout: ${build.stdout?.toString().trim()}\n\n\nstderr: ${build.stderr?.toString().trim()}`);
       }
 
-      const args = [
-        'build', this.options.entry,
-        '--out-dir', this.options.outDir,
+      // Find the git root and mount it in the container. This allows Parcel to
+      // find the same modules/dependencies as the ones available "locally". It
+      // also supports monorepos.
+      const projectRoot = path.dirname(findGitPath());
```

Comment:
@eladb this is the fix

Reply:
This implies that the project root is the git root which means:
1. That the project directory must be inside a git repo. This is not always the case. For example, last time I checked, when you used GitHub as a source for CodePipeline, the CodeBuild project did not include a git clone but rather a copy of the source tree (without a .git directory).
2. Maybe we need to allow people to configure this?

Reply:
Do you have maybe a suggestion to find the root of a monorepo?

Reply:
How about we allow people to specify it and default to searching for `.git`, but if we can't find we omit an error explaining how to manually configure.